duffel_api/models/offer.py

```diff
@@ -221,6 +221,32 @@ def from_json(cls, json: dict):
         )
 
 
+@dataclass
+class OfferSliceSegmentStop:
+    """Additional segment-specific information about the stops, if any, included in the segment"""
+
+    id: str
+    duration: str
+    arriving_at: datetime
+    departing_at: datetime
+    airport: Airport
+
+    @classmethod
+    def from_json(cls, json: dict):
+        """Construct a class instance from a JSON response."""
+        return cls(
+            id=json["id"],
+            arriving_at=get_and_transform(
+                json, "arriving_at", parse_datetime
+            ),
+            departing_at=get_and_transform(
+                json, "departing_at", parse_datetime
+            ),
```

Comment:
**fix:** The reason for the static analysis (`pyright`) failures is that this helper function, `get_and_transform`, is for when values are optional.

Those 2 aren't, so you can parse them directly.

```suggestion
            arriving_at=parse_datetime(json["arriving_at"]),
            departing_at=parse_datetime(json["departing_at"]),
```

Reply:
Ahh sorry, it was my bad